tests/extras/datasets/pandas/test_excel_dataset.py

```diff
@@ -151,6 +151,7 @@ def test_protocol_usage(self, filepath, instance_type, load_path, mocker):
         assert isinstance(data_set._filepath, PurePosixPath)
 
         mock_pandas_call = mocker.patch("pandas.read_excel")
+        mocker.patch("kedro.io.core.AbstractVersionedDataSet._filesystem_exists")
```

Comment:
The test is for testing the DataSet implementation so it doesn't feel right to mock the implementation. What's the error if you didn't mock it?

Reply:
The datasets used for these tests are dummies and in the original version of the tests, the actual `pandas.read_excel` (and the respective read calls for the different pandas datasets) calls are mock calls. These tests were failing due to the changes in this PR because the `_filesystem_exists()` call tries to check if these datasets actually exist.